code/modules/projectiles/ammunition/ballistic/rifle.dm

```diff
@@ -73,51 +81,83 @@
 
 /obj/item/ammo_casing/rebar/zaukerite
-	name = "Zaukerite Sliver"
+	name = "zaukerite sliver"
 	desc = "A sliver of a zaukerite crystal. Due to its irregular, jagged edges, removal of an embedded zaukerite sliver should only be done by trained surgeons."
 	caliber = CALIBER_REBAR
 	icon_state = "rod_zaukerite"
 	base_icon_state = "rod_zaukerite"
 	projectile_type = /obj/projectile/bullet/rebar/zaukerite
 
 /obj/item/ammo_casing/rebar/hydrogen
-	name = "Metallic Hydrogen Bolt"
+	name = "metallic hydrogen bolt"
 	desc = "An ultra-sharp rod made from pure metallic hydrogen. Armor may as well not exist."
 	caliber = CALIBER_REBAR
 	icon_state = "rod_hydrogen"
 	base_icon_state = "rod_hydrogen"
 	projectile_type = /obj/projectile/bullet/rebar/hydrogen
 
 /obj/item/ammo_casing/rebar/healium
-	name = "Healium Crystal Bolt"
+	name = "healium crystal bolt"
 	desc = "Who needs a syringe gun, anyway?"
 	caliber = CALIBER_REBAR
 	icon_state = "rod_healium"
 	base_icon_state =  "rod_healium"
 	projectile_type = /obj/projectile/bullet/rebar/healium
+	/// How many seconds of healing/sleeping action we have left, once all are spent the bolt dissolves
+	var/heals_left = 6 SECONDS
+
+/obj/item/ammo_casing/rebar/healium/ready_proj(atom/target, mob/living/user, quiet, zone_override, atom/fired_from)
+	if (loaded_projectile)
+		var/obj/projectile/bullet/rebar/healium/bolt = loaded_projectile
+		bolt.heals_left = heals_left
+	return ..()
+
+/datum/embedding/rebar_healium
+	embed_chance = 100
+	fall_chance = 0
+	pain_stam_pct = 0.9
+	ignore_throwspeed_threshold = TRUE
+	rip_time = 1.5 SECONDS
+	/// Amount of each type of damage healed per second
+	var/healing_per_second = 5
+	/// Amount of drowsiness added per second
+	var/drowsy_per_second = 2 SECONDS
+	/// At what point of drowsiness do we knock out the owner
+	var/drowsy_knockout = 5 SECONDS // Actually more like 80, because you need 4 ticks to reach this
+
+/datum/embedding/rebar_healium/process(seconds_per_tick)
+	. = ..()
+	var/obj/item/ammo_casing/rebar/healium/casing = parent
+	casing.heals_left -= seconds_per_tick * 1 SECONDS
+	var/update_health = FALSE
+	var/healing = healing_per_second * seconds_per_tick
+	update_health ||= owner.adjustBruteLoss(healing, updating_health = FALSE, required_bodytype = BODYTYPE_ORGANIC)
+	update_health ||= owner.adjustFireLoss(healing, updating_health = FALSE, required_bodytype = BODYTYPE_ORGANIC)
+	update_health ||= owner.adjustToxLoss(healing, updating_health = FALSE, required_biotype = BODYTYPE_ORGANIC)
+	update_health ||= owner.adjustOxyLoss(healing, updating_health = FALSE, required_biotype = BODYTYPE_ORGANIC)
+	if (update_health)
+		owner.updatehealth()
+	if (owner.mob_biotypes & MOB_ORGANIC)
+		owner.adjust_drowsiness(drowsy_per_second * seconds_per_tick)
+		var/datum/status_effect/drowsiness/drowsiness = owner.has_status_effect(/datum/status_effect/drowsiness)
+		if (drowsiness?.duration - world.time >= drowsy_knockout)
+			owner.SetSleeping(3 SECONDS)
```

Comment:
Using `SetSleeping` means this can technically "heal" sleeping (reducing the duration of their current sleep)